android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java

```diff
@@ -2093,7 +2100,66 @@ private void handleErrorsFromInternalMessageDispatcher(String info, Exception e)
 		_proxyListener.onError("Proxy callback dispatcher is down. Proxy instance is invalid.", e);
 	}
 	/************* END Functions used by the Message Dispatching Queues ****************/
-	
+
+	private OnRPCNotificationListener mEncryptionReqListener = new OnRPCNotificationListener() {
+		@Override
+		public void onNotified(RPCNotification notification) {
+			List<PermissionItem> permissionItems = ((OnPermissionsChange)notification).getPermissionItem();
+			Set<String> encryptedRpcs = new HashSet<>();
+			boolean requireEncryptionAppLevel = Boolean.TRUE.equals(((OnPermissionsChange) notification).getEncryptionRequirement());
+			for (PermissionItem permissionItem : permissionItems) {
+				if (permissionItems != null && !permissionItems.isEmpty()) {
+					if (permissionItem != null) {
+					    if (requireEncryptionAppLevel && Boolean.TRUE.equals(permissionItem.getEncryptionRequirement())) {
+							String rpcName = permissionItem.getRpcName();
+							if (rpcName != null) {
+								encryptedRpcs.add(rpcName);
+							}
+						}
+					}
+				}
+			}
+			if (!encryptedRpcs.isEmpty()) {
+				if (!mRPCSecuredServiceStarted) {
+					startProtectedRPCService();
+				}
+				if (!mEncryptedRPCNames.equals(encryptedRpcs)) {
+					mEncryptedRPCNames = encryptedRpcs;
+				}
+			}
+		}
+	};
+
+	private ISdlServiceListener mRPCSecuredServiceListener = new ISdlServiceListener() {
+		@Override
+		public void onServiceStarted(SdlSession session, SessionType type, boolean isEncrypted) {
+			if(SessionType.RPC.equals(type) && session != null ){

```

Comment:
Should we add a check in the if statement to check isEncrypted is true? Possibly adds a touch more efficiency not setting mRPCSecuredServiceStarted when not necessary.

Reply:
Thanks Markos, added the conditional statement for `isEncrypted`.